modules/basegl/src/processors/imageprocessing/imagesubsetgl.cpp

```diff
@@ -0,0 +1,239 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/processors/imageprocessing/imagesubsetgl.h>
+#include <inviwo/core/datastructures/buffer/buffer.h>
+#include <inviwo/core/interaction/events/gestureevent.h>
+#include <inviwo/core/interaction/events/mouseevent.h>
+#include <inviwo/core/interaction/events/touchevent.h>
+#include <inviwo/core/network/networklock.h>
+
+#include <modules/opengl/geometry/meshgl.h>
+#include <modules/opengl/openglutils.h>
+#include <modules/opengl/texture/textureutils.h>
+
+namespace inviwo {
+
+const ProcessorInfo ImageSubsetGL::processorInfo_{
+    "org.inviwo.ImageSubsetGL",  // Class identifier
+    "Image Subset",              // Display name
+    "Image Operation",           // Category
+    CodeState::Stable,           // Code state
+    Tags::GL,                    // Tags
+};
+const ProcessorInfo ImageSubsetGL::getProcessorInfo() const { return processorInfo_; }
+
+ImageSubsetGL::ImageSubsetGL()
+    : Processor()
+    , inport_("inputImage")
+    , outport_("outputImage")
+    , rangeX_("rangeX", "X Slices", 0, 256, 0, 256, 1, 1)
+    , rangeY_("rangeY", "Y Slices", 0, 256, 0, 256, 1, 1)
+    , shader_("img_texturequad.vert", "img_copy.frag") {
+    addPort(inport_);
+    addPort(outport_);
+    addProperties(rangeX_, rangeY_);
+    dims_ = size2_t(1, 1);
+
+    inport_.setOutportDeterminesSize(true);
+    outport_.setHandleResizeEvents(false);
+    // Since the ranges depend on the input image dimensions, we make sure to always
+    // serialize them so we can do a proper renormalization when we load new data.
+    rangeX_.setSerializationMode(PropertySerializationMode::All);
+    rangeY_.setSerializationMode(PropertySerializationMode::All);
+
+    inport_.onChange([this]() {
+        NetworkLock lock(this);
+
+        // Update to the new dimensions.
+        dims_ = inport_.getData()->getDimensions();
+
+        rangeX_.setRangeNormalized(size2_t(0, dims_.x));
+        rangeY_.setRangeNormalized(size2_t(0, dims_.y));
+
+        // set the new dimensions to default if we were to press reset
+        rangeX_.setCurrentStateAsDefault();
+        rangeY_.setCurrentStateAsDefault();
+    });
+
+    // Create a rectangle mesh for rendering.
+    // Texture coordinates will be modified
+    auto verticesBuffer =
+        util::makeBuffer<vec2>({{-1.0f, -1.0f}, {1.0f, -1.0f}, {-1.0f, 1.0f}, {1.0f, 1.0f}});
+    vec2 texOffset = vec2(0.f) / vec2(dims_);
+    vec2 texEnd = vec2(1.f) / vec2(dims_);
+    texCoordsBuffer_ = util::makeBuffer<vec2>(
+        {texOffset, {texEnd.x, texOffset.y}, {texOffset.x, texEnd.y}, texEnd});
+    auto indices = util::makeIndexBuffer({0, 1, 2, 3});
+    rect_.addBuffer(BufferType::PositionAttrib, verticesBuffer);
```

Comment:
Could use a typed mesh instead: `TypedMesh<buffertraits::PositionsBuffer2D, buffertraits::TexcoordBuffer<2>>`

Reply:
Ok, just copied the code in drawImagePlaneRect :)

Reply:
done